Code/Framework/AzCore/AzCore/Math/MatrixUtils.cpp

```diff
@@ -60,6 +60,20 @@ namespace AZ
         return 2.0f * AZStd::atan(1.0f / m.GetElement(1, 1));
     }
 
+    void SetPerspectiveMatrixNearFar(Matrix4x4& out, float nearDist, float farDist, bool reverseDepth)
+    {
+        AZ_Assert(nearDist > FloatEpsilon, "Near distance should be greater than zero (float epsilon)");
+        AZ_Assert(farDist > nearDist, "Far should be greater than near");
```

Comment:
nit: Can we add a little more context to the assert? I'm not sure if this message would just show on its own, so specifying it's a Perspective Matrix error could help :)

Reply:
Here you are